main/modes/games/bigbug/mode_bigbug.c

```diff
@@ -480,32 +477,22 @@ static void bb_BackgroundDrawCallback(int16_t x, int16_t y, int16_t w, int16_t h
             bool drawGrRow             = (0 <= grRow && grRow < tilemap->landfillGradient.h);
             paletteColor_t* gradientPx = &tilemap->landfillGradient.px[grWidth * grRow];
 
-            // Start indices at X offsets
-            int32_t s1idx = offsetX1;
-            int32_t s2idx = offsetX2;
-            int32_t gridx = offsetXG;
-
             // For each pixel in the row
             for (int32_t iX = x; iX < x + w; iX++)
             {
                 // Draw appropriate pixel
-                if (drawS1row && cTransparent != surface1px[s1idx])
+                if (drawS1row && cTransparent != surface1px[iX])
                 {
-                    TURBO_SET_PIXEL(iX, iY, surface1px[s1idx]);
+                    *(fb++) = surface1px[iX];
```

Comment:
This ends up being faster than `TURBO`, I think, because `TURBO` does a bunch of X/Y math to draw the pixel, but this is a special case where we're filling the entire background, so we can simply step the framebuffer pointer (a single addition).